pyswarms/backend/handlers.py

```diff
@@ -0,0 +1,163 @@
+        in the next optimization step by using one of the follwing methods:
+
+        * Nearest:
+            Reposition the particle to the nearest bound.
+        * Random:
+            Reposition the particle randomly in between the bounds.
+        * Shrink:
+            Shrink the velocity of the particle such that it lands on the
+            bounds.
+        * Reflective:
+            Mirror the particle position from outside the bounds to inside the
+            bounds.
+        * Intermediate:
+            Reposition the particle to the midpoint between its current
+            position on the bound surpassing axis and the bound itself.
+            This only adjusts the axes that surpass the boundaries.
+        * Resample:
+            Redraw the velocity until the next position is inside the bounds.
+
+        Attributes
+        ----------
+        strategy : str
+            The strategy to be used. The following are available:
+            * "nearest"
+            * "random"
+            * "shrink"
+            * "reflective"
+            * "intermediate"
+            * "resample"
+            For a description of these see above.
+        """
+        self.strategy = strategy
+
+    def __call__(self, position, bounds, *args, **kwargs):
+        """Make class callable
+
+        The BoundaryHandler can be called as a function to use the strategy
+        that is passed at initialization to repair boundary issues. An example
+        for the usage:
+
+        .. code-block :: python
+
+            from pyswarms.backend import operators as op
+            from pyswarms.backend.handlers import BoundaryHandler
+
+            bh = BoundaryHandler(strategy="reflective")
+            ops.compute_position(swarm, bounds, handler=bh)
+
+
+        Parameters
+        ----------
+        position : np.ndarray
+            The swarm position to be handled
+        bounds : tuple of :code:`np.ndarray` or list
+            a tuple of size 2 where the first entry is the minimum bound while
+            the second entry is the maximum bound. Each array must be of shape
+            :code:`(dimensions,)`
+        *args : tuple
+        **kwargs : dict
+
+        Returns
+        -------
+        numpy.ndarray
+            the adjusted positions of the swarm
+        """
+        # Assign new attributes
+        self.position = position
+        self.lower_bound, self.upper_bound = bounds
+        self.__out_of_bounds()
+
+        if self.strategy == "nearest":
+            new_position = self.nearest()
+        elif self.strategy == "reflective":
+            new_position = self.reflective()
+        elif self.strategy == "shrink":
+            new_position = self.shrink()
+        elif self.strategy == "random":
+            new_position = self.random()
+        elif self.strategy == "intermediate":
+            new_position = self.random()
+        elif self.strategy == "resample":
+            new_position = self.resample()
```

Comment:
I'm wondering if it's better to use a dictionary for pattern matching...

```python
x = {"nearest": self.nearest, "resample": self.resample}
x(position, bounds, *args, **kwargs)
```

Reply:
Wouldn't it be x["nearest"](position, bounds, *args, **kwargs) then?

Reply:
Yup you are correct